src/drivers/drv_hrt.h

```diff
@@ -149,8 +149,8 @@ static inline hrt_abstime ts_to_abstime(const struct timespec *ts)
  */
 static inline void abstime_to_ts(struct timespec *ts, hrt_abstime abstime)
 {
-	ts->tv_sec = (time_t)abstime / 1000000;
-	abstime -= (hrt_abstime)(ts->tv_sec * 1000000);
```

Comment:
first multiplying 32-bit value with 1000000 overflows, and subtracting that from abstime (64 bits) causes result being larger than a second (>1000000), which causes invalid value assingment on the next line.